bin/internal/update_dart_sdk.sh

```diff
@@ -116,6 +116,13 @@ if [ ! -f "$ENGINE_STAMP" ] || [ "$ENGINE_VERSION" != `cat "$ENGINE_STAMP"` ]; t
   fi
 
   curl ${verbose_curl} --retry 3 --continue-at - --location --output "$DART_SDK_ZIP" "$DART_SDK_URL" 2>&1 || {
+    curlExitCode=$?
+    if [ $curlExitCode != 33 ]; then
```

Comment:
can you add a comment here explaining what 33 means and what situations we've seen it in and why we're doing something special with it and so on?